### PR TITLE
support tcp transport protocol

### PR DESCRIPTION
**Pull request by colinsurprenant**
Add support for tcp transport protocol.

I locally tested this both for udp and tcp and both are working. We are lacking test/spec coverage for this and I will followup with an initial spec for the client.

### CODE REVIEW

File: spec/inputs/snmp_spec.rb [hunk 1]
```diff
@@ -54,13 +54,15 @@
           {"get" => ["1.0"], "hosts" => [{"host" => "udp:localhost/161"}]},
           {"get" => ["1.0"], "hosts" => [{"host" => "udp:127.0.0.1/112345"}]},
           {"get" => ["1.0"], "hosts" => [{"host" => "udp:127.0.0.1/161", "community" => "public"}]},
+          {"get" => ["1.0"], "hosts" => [{"host" => "tcp:127.0.0.1/112345"}]},
+          {"get" => ["1.0"], "hosts" => [{"host" => "tcp:127.0.0.1/161", "community" => "public"}]},
```
Comment (jsvd) [inline spec/inputs/snmp_spec.rb]: should't this be 1161?

Comment (colinsurprenant) [reply spec/inputs/snmp_spec.rb]: doesn't have to be - note that we are only verifying the format here but in any case, the port number can be any number.

Comment (jsvd) [reply spec/inputs/snmp_spec.rb]: right, I was just aiming at consistency in the examples that reference 1161 for tcp in two other places.

**Comment by colinsurprenant** [comment]
Thanks @jsvd ! will be pushing 0.1.0.beta3 shortly.

**Pull request merged**